### PR TITLE
Fix/help page mobile enhancement (issue #105)

### PR DESCRIPTION
**Pull request by Astrejoe**
#### Deployment instructions
/

#### What have I done
Resolved issue #105
I enhanced the help-page on mobile view. This includes making it more responsive, centering the content on mobile, creating a different view for tablets and adding dividers between the key description. Internally I also added a key-description component in order to make the code more DRY.

#### Testing
Resize the help page from full desktop view all the way down to mobile view. On every size it should provide a pleasant layout.

### CODE REVIEW

File: src/app/help-page/help-page.component.html [hunk 1]
```diff
@@ -18,31 +18,31 @@ <h1 class="d-none d-md-block">Help</h1><br/>
...
-                <div class="row no-gutters pb-3">
-                    <div class="col-md-4"><img class="key-large" src="/assets/keys/Enter.png" alt="enter"></div>
-                    <div class="col-md-8">Choose the printer system you want to visualize and import it.</div>
-                </div>
-                <div class="row no-gutters pb-3">
-                    <div class="col-md-4"><img class="key-small" src="/assets/keys/5.png" alt="num 5"></div>
-                    <div class="col-md-8">Choose the printer system you want to visualize and import it.</div>
-                </div>
+                <h3 class="pt-3 text-center text-md-left">Desktop</h3><hr>
+                <app-key-description [source]="'/assets/keys/Enter.png'" [alternative]="'enter key'"
```
Comment (Dacaspex) [inline src/app/help-page/help-page.component.html]: Looks good! I would change it a bit code style wise to make it a bit easier to read
```
<app-key-description
    [source]=""
    [alternative]=""
    [description]=""
></app-key-description>
```
So you indent the `[property]` with a tab instead of one space, and each property on a separate line

File: src/app/help-page/help-page.component.ts [hunk 2]
```diff
@@ -1,5 +1,4 @@
 import { Component, OnInit } from '@angular/core';
-import { faArrowLeft } from '@fortawesome/free-solid-svg-icons';
```
Comment (Dacaspex) [inline src/app/help-page/help-page.component.ts]: Could you update the indentation of this file to four spaces?

File: src/app/help-page/key-description/key-description.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (Dacaspex) [inline src/app/help-page/key-description/key-description.component.spec.ts]: Could you fix the indentation to four spaces? You can set this default in Webstorm. I can help you fix that if you'd like

File: src/app/help-page/key-description/key-description.component.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { Component, Input, OnInit } from '@angular/core';
+
+@Component({
+  selector: 'app-key-description',
+  templateUrl: './key-description.component.html',
+  styleUrls: ['./key-description.component.scss']
+})
+export class KeyDescriptionComponent implements OnInit {
```
Comment (Dacaspex) [inline src/app/help-page/key-description/key-description.component.ts]: - Could you fix the indentation?
- I would add a comment at the `@Input()` variables, what do they mean / what are they used for?

File: src/app/menubar/menubar.component.html [hunk 5]
```diff
@@ -21,12 +21,18 @@
             class="cursor-pointer"
         >
                 <fa-icon [icon]="arrowLeftIcon"> </fa-icon>
-                <span class="ml-2 d-none d-lg-inline-block"> Back to workspace </span>
+                <span class="ml-2 d-none d-xl-inline-block"> Back to workspace </span>
```
Comment (Dacaspex) [inline src/app/menubar/menubar.component.html]: I think we can also show this text on `lg`, is there a specific reason why you didn't?

File: src/app/menubar/menubar.component.html [hunk 6]
```diff
@@ -43,7 +49,7 @@ <h3 *ngIf="inHelp()">Help</h3>
     On mobile, this is the VR, overflow-menu and settings button
     -->
     <app-help
-        *ngIf="inWorkspace()"
+        [style.visibility]="inWorkspace() ? 'visible' : 'hidden'"
```
Comment (Dacaspex) [inline src/app/menubar/menubar.component.html]: Why is this changed from `*ngif=inWorkspace()`? That is how we do it in most places. Was there a specific reason to do this instead?

Comment (Astrejoe) [reply src/app/menubar/menubar.component.html]: Yes. I did this in order to center the Help and Settings titles in the menu bar on mobile. The menu bar is a bootstrap-row. On the left and on the right of the title, there are col-2's, while the title itself is a col-8. Using *ngIf, there are cases where there is no col-2 on the right side of the title. This gives the title itself an improper width.

File: src/app/menubar/menubar.component.html (hunk 5 above)
Comment (Astrejoe) [reply src/app/menubar/menubar.component.html]: Yes, I was also hesitant about this one. I basically did it for consistency. Now _back to workspace_ appears and disappears together with the image on the right side. This felt better to me.

File: src/app/help-page/key-description/key-description.component.ts (hunk 4 above)
Comment (Astrejoe) [reply src/app/help-page/key-description/key-description.component.ts]: done

File: src/app/help-page/key-description/key-description.component.spec.ts (hunk 3 above)
Comment (Astrejoe) [reply src/app/help-page/key-description/key-description.component.spec.ts]: done

File: src/app/help-page/help-page.component.ts (hunk 2 above)
Comment (Astrejoe) [reply src/app/help-page/help-page.component.ts]: done

File: src/app/help-page/help-page.component.html (hunk 1 above)
Comment (Astrejoe) [reply src/app/help-page/help-page.component.html]: done

File: src/app/menubar/menubar.component.html (hunk 5 above)
Comment (Dacaspex) [reply src/app/menubar/menubar.component.html]: Sounds fair!

File: src/app/menubar/menubar.component.html (hunk 6 above)
Comment (Dacaspex) [reply src/app/menubar/menubar.component.html]: Good point :+1:
